accounting/invoice/views.py

```diff
@@ -20,3 +20,47 @@
 """
 Invoice views.
 """
+
+from rest_framework import decorators, permissions, response, viewsets
+
+from accounting.invoice import choices, models
+
+
+class InvoiceViewSet(viewsets.ModelViewSet):
+    """
+    A view for retrieving and updating `Invoice` objects.
+    """
+
+    permission_classes = permissions.IsAdminUser
+    queryset = models.Invoice.objects.all()
+    lookup_field = 'uuid'
+
+    @decorators.detail_route(
+        permission_classes=[permissions.AllowAny],
```

Comment:
@UmanShahzad Just to confirm, this endpoint is still protected by the class' `permission_classes` (permission.IsAdminUser), right?

Reply:
Only these are protected by `IsAdminUser`:

1. /invoice/
1. /invoice/{uuid}
1. /invoice/{uuid}/pay/

We need `/invoice/{uuid}/approve/` to be open since otherwise no one could confirm their invoice.

Reply:
@UmanShahzad Ack, of course -- for some reason I thought I was looking at the `pay` endpoint. Sorry for the noise!

Reply:
:+1: